server/src/main/java/io/druid/curator/CuratorModule.java

```diff
@@ -91,6 +101,54 @@ public void stop()
     return framework;
   }
 
+  private CuratorFrameworkFactory.Builder newCuratorBuilder(CuratorConfig config, ExhibitorConfig exConfig)
+  {
+    CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder();
+    Optional<EnsembleProvider> provider = newEnsembleProvider(exConfig, config.getZkHosts());
+    if (provider.isPresent()) {
+      return builder.ensembleProvider(provider.get());
+    }
+    return builder.connectString(config.getZkHosts());
+  }
+
+  private Optional<EnsembleProvider> newEnsembleProvider(ExhibitorConfig config, String backupZKHosts)
+  {
+    List<String> exhibitorHosts = config.getHosts();
+    if (exhibitorHosts.isEmpty()) {
+      return Optional.absent();
+    }
+    ExhibitorEnsembleProvider provider = new ExhibitorEnsembleProvider(
+        new Exhibitors(
+            exhibitorHosts,
+            config.getRestPort(),
+            newBackupProvider(backupZKHosts)
+        ),
+        new DefaultExhibitorRestClient(config.getUseSsl()),
+        config.getRestUriPath(),
+        config.getPollingMs(),
+        new BoundedExponentialBackoffRetry(1000, 45000, 10)
+    );
+    try {
+      provider.pollForInitialEnsemble();
+      return Optional.<EnsembleProvider>of(provider);
+    }
+    catch (Exception err) {
```

Comment:
It is not clear to me this should succeed. For example, if the polling is stuck and there's an `InterruptedException`, or a `RuntimeException` that wraps the interrupted information, this should fail.

Is there a reason the initial polling needs to take place?